libs/types/src/cdf/ballot-definition/convert.ts

```diff
@@ -30,6 +31,326 @@ function officeId(contestId: Vxf.ContestId): string {
+              style: 'narrow',
+              fallback: 'none',
+            }).of(languageCode)
+          ),
+        })
+      ),
+    });
+  },
+
+  [ElectionStringKey.BALLOT_STYLE_ID](cdfElection, uiStrings) {
+    for (const ballotStyle of assertDefined(cdfElection.Election[0])
+      .BallotStyle) {
+      const ballotStyleId = assertDefined(
+        ballotStyle.ExternalIdentifier[0]
+      ).Value;
+
+      setStaticUiString({
+        stringKey: [ElectionStringKey.BALLOT_STYLE_ID, ballotStyleId],
+        uiStrings,
+        // TODO(kofi): Should we start populating the `Label` field to provide
+        // more user-friendly display values?
+        value: assertDefined(ballotStyle.ExternalIdentifier[0]).Value,
+      });
+    }
+  },
+
+  [ElectionStringKey.CANDIDATE_NAME](cdfElection, uiStrings) {
+    const candidates = assertDefined(cdfElection.Election[0]).Candidate || [];
+    for (const candidate of candidates) {
+      setInternationalizedUiStrings({
+        stringKey: [ElectionStringKey.CANDIDATE_NAME, candidate['@id']],
+        uiStrings,
+        values: candidate.BallotName.Text,
+      });
+    }
+  },
+
+  [ElectionStringKey.CONTEST_DESCRIPTION](cdfElection, uiStrings) {
+    for (const contest of assertDefined(cdfElection.Election[0]).Contest) {
+      if (contest['@type'] !== 'BallotDefinition.BallotMeasureContest') {
+        continue;
+      }
+
+      setInternationalizedUiStrings({
+        stringKey: [ElectionStringKey.CONTEST_DESCRIPTION, contest['@id']],
+        uiStrings,
+        values: contest.FullText.Text,
+      });
+    }
+  },
+
+  [ElectionStringKey.CONTEST_OPTION_LABEL](cdfElection, uiStrings) {
+    for (const contest of assertDefined(cdfElection.Election[0]).Contest) {
+      if (contest['@type'] !== 'BallotDefinition.BallotMeasureContest') {
+        continue;
+      }
+
+      for (const option of contest.ContestOption) {
+        setInternationalizedUiStrings({
+          stringKey: [ElectionStringKey.CONTEST_OPTION_LABEL, option['@id']],
+          uiStrings,
+          values: option.Selection.Text,
+        });
+      }
+    }
+  },
+
+  [ElectionStringKey.CONTEST_TERM](cdfElection, uiStrings) {
+    // CDF does not support internationalized text for this string, so we just
+    // use the provided English text.
+    for (const contest of assertDefined(cdfElection.Election[0]).Contest) {
+      const termDescription = termDescriptionForContest(
+        cdfElection,
+        contest['@id']
+      );
+      if (termDescription) {
+        setInternationalizedUiStrings({
+          stringKey: [ElectionStringKey.CONTEST_TERM, contest['@id']],
+          uiStrings,
+          values: electionLanguageCodes(cdfElection).map(
+            (languageCode): Cdf.LanguageString => ({
+              '@type': 'BallotDefinition.LanguageString',
+              Language: languageCode,
+              Content: termDescription,
+            })
+          ),
+        });
+      }
+    }
+  },
```

Comment:
Similarly, I needed a default for term description since it doesn't have internationalization support in CDF. In this case, JS doesn't have any native way to translate the arbitrary string, so I couldn't actually do anything but use the English. It's not really a passable solution, so I would say we just don't support term description translations when using CDF, which seems ok since CDF itself doesn't support them.

Reply:
Yeah, a warning sounds like a good idea.

> If they import CDF then they should still be able to add term afterwords right?
In VxDesign yes, but not in the rest of VxSuite